jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py

```diff
@@ -205,30 +216,43 @@ def collapse_to_spectrum(self, add_data=True, **kwargs):
         """
         spectral_cube = self._app._jdaviz_helper._loaded_flux_cube
         uncert_cube = self._app._jdaviz_helper._loaded_uncert_cube
+        uncertainties = None
 
         # This plugin collapses over the *spatial axes* (optionally over a spatial subset,
         # defaults to ``No Subset``). Since the Cubeviz parser puts the fluxes
         # and uncertainties in different glue Data objects, we translate the spectral
         # cube and its uncertainties into separate NDDataArrays, then combine them:
-        if self.aperture.selected != self.aperture.default_text:
+        if self.aperture.selected != self.aperture.default_text and self.wavelength_dependent:
             nddata = spectral_cube.get_subset_object(
                 subset_id=self.aperture.selected, cls=NDDataArray
             )
-            uncertainties = uncert_cube.get_subset_object(
-                subset_id=self.aperture.selected, cls=StdDevUncertainty
+            if uncert_cube:
+                uncertainties = uncert_cube.get_subset_object(
+                    subset_id=self.aperture.selected, cls=StdDevUncertainty
+                )
+            mask = self.cone_aperture()
+
+        elif self.aperture.selected != self.aperture.default_text:
+            nddata = spectral_cube.get_subset_object(
+                subset_id=self.aperture.selected, cls=NDDataArray
             )
+            if uncert_cube:
+                uncertainties = uncert_cube.get_subset_object(
+                    subset_id=self.aperture.selected, cls=StdDevUncertainty
```

Comment:
I think this is wrong unless the uncertainty type is normalized outside of the parser code. Here we allow at least 3 different types of uncertainty (inverse variance, variance, standard deviation):

https://github.com/spacetelescope/jdaviz/blob/07de88325a799004a111cdfb9454a52b6e55a89a/jdaviz/configs/cubeviz/plugins/parsers.py#L20

But looks like we just parse those in blindly as-is. How do we know for sure at this point that the uncertainty is always standard deviation?

Reply:
To this effect, we also need to double check that mask values are being used as intended. Usually telescope ship the MASK as DQ arrays on bitplane, not pure boolean. For display purpose, does not matter, but now that you are passing it into some sort of extraction, how is this mask actually used? Not all "bad" pixels have the same level of badness. Here are example bitplane values.

https://github.com/spacetelescope/stginga/blob/master/stginga/data/dqflags_jwst.txt

Reply:
I opened 2 follow-up issues:

* https://github.com/spacetelescope/jdaviz/issues/2713
* https://github.com/spacetelescope/jdaviz/issues/2714